src/ray/util/tests/pipe_logger_test.cc

```diff
@@ -132,50 +109,91 @@ TEST(PipeLoggerTestWithTee, RedirectionWithTee) {
-  logging_option.file_path = test_file_path;
-  logging_option.rotation_max_size = 5;
-  logging_option.rotation_max_file_count = 2;
-  logging_option.tee_to_stderr = true;
+  StreamRedirectionOption stream_redirection_opt{};
+  stream_redirection_opt.file_path = test_file_path;
+  stream_redirection_opt.rotation_max_size = 5;
+  stream_redirection_opt.rotation_max_file_count = 2;
+  stream_redirection_opt.tee_to_stderr = true;
 
   // Capture stdout via `dup`.
   testing::internal::CaptureStderr();
 
-  auto log_token = CreateRedirectionFileHandle(logging_option);
-  ASSERT_EQ(write(log_token.GetWriteHandle(), kLogLine1.data(), kLogLine1.length()),
-            kLogLine1.length());
-  ASSERT_EQ(write(log_token.GetWriteHandle(), kLogLine2.data(), kLogLine2.length()),
-            kLogLine2.length());
-  log_token.Close();
+  auto stream_redirection_handle = CreateRedirectionFileHandle(stream_redirection_opt);
+  stream_redirection_handle.CompleteWrite(kLogLine1.data(), kLogLine1.length());
+  stream_redirection_handle.CompleteWrite(kLogLine2.data(), kLogLine2.length());
+  stream_redirection_handle.Close();
 
   // Check content tee-ed to stderr.
   const std::string stderr_content = testing::internal::GetCapturedStderr();
   EXPECT_EQ(stderr_content, absl::StrFormat("%s%s", kLogLine1, kLogLine2));
 
   // Check log content after completion.
-  const std::string log_file_path1 = test_file_path;
   EXPECT_EQ(CompleteReadFile(test_file_path), kLogLine2);
-
-  const std::string log_file_path2 = absl::StrFormat("%s.1", test_file_path);
   EXPECT_EQ(CompleteReadFile(log_file_path2), kLogLine1);
-
-  // Delete temporary file.
-  EXPECT_EQ(unlink(log_file_path1.data()), 0);
-  EXPECT_EQ(unlink(log_file_path2.data()), 0);
 }
 
 }  // namespace
 
 }  // namespace ray
 
 #endif
+
+/////////////////////////////////////////////////
+// Unit test for both windows and unix platform.
+/////////////////////////////////////////////////
+
+namespace ray {
+
+namespace {
+
+class PipeLoggerWithRedirection : public ::testing::TestWithParam<size_t> {};
+
+TEST_P(PipeLoggerWithRedirection, NoPipeWrite) {
+  const size_t pipe_buffer_size = GetParam();
+  setenv(kPipeLogReadBufSizeEnv.data(),
+         absl::StrFormat("%d", pipe_buffer_size).data(),
+         /*overwrite=*/1);
+
+  // TODO(hjiang): We should have a better test util, which allows us to create a
+  // temporary testing directory.
+  const std::string test_file_path = absl::StrFormat("%s.out", GenerateUUIDV4());
+
+  // Delete temporary file.
+  absl::Cleanup cleanup_test_file = [&test_file_path]() {
+    EXPECT_TRUE(std::filesystem::remove(test_file_path));
+  };
+
+  // Take the default option, which doesn't have rotation enabled.
+  StreamRedirectionOption stream_redirection_opt{};
+  stream_redirection_opt.file_path = test_file_path;
+  auto stream_redirection_handle = CreateRedirectionFileHandle(stream_redirection_opt);
+  stream_redirection_handle.CompleteWrite(kLogLine1.data(), kLogLine1.length());
```

Comment:
Currently kLogLine1 is defined inside `#if defined(__APPLE__) || defined(__linux__)` so this won't compile on Windows?

Reply:
updated